scripts/create_multi_session_df.py

```diff
@@ -19,90 +19,89 @@
+                         ['cell_specimen_id', 'is_change', 'image_name', 'epoch'],
+                         ['cell_specimen_id', 'is_change', 'hit'],
+                         ['cell_specimen_id', 'pre_change', 'epoch'],
+                         ['cell_specimen_id', 'is_change', 'hit', 'epoch'],
+                         ['cell_specimen_id', 'omitted', 'pre_omitted'],]
+
+    behavior_conditions = [['ophys_experiment_id', 'omitted'],
+                            ['ophys_experiment_id', 'omitted', 'epoch'],
+                            ['ophys_experiment_id', 'is_change'],
+                            ['ophys_experiment_id', 'is_change', 'epoch'],
+                            ['ophys_experiment_id', 'is_change', 'image_name'],
+                            ['ophys_experiment_id', 'is_change', 'image_name', 'epoch'],
+                            ['ophys_experiment_id', 'is_change', 'hit'],
+                            ['ophys_experiment_id', 'is_change', 'pre_change', 'epoch'],
+                            ['ophys_experiment_id', 'is_change', 'hit', 'epoch'],
+                            ['cell_specimen_id', 'omitted', 'pre_omitted'],]
+
+
+    # event types corresponding to the above physio and behavior conditions - must be in same sequential order
+    event_types_for_conditions = ['omissions', 'omissions',
+                                  'changes', 'changes',
+                                  'changes', 'changes', 'changes',
+                                  'all', 'all', 'all']
+
+    # add engagement state to all conditions
+    # for i in range(len(physio_conditions)):
+    #     physio_conditions[i].insert(1, 'engagement_state')
+    #     behavior_conditions[i].insert(1, 'engagement_state')
+
+    # create dfs for all data types and conditions for physio data
+    for data_type in physio_data_types:
+    # data_type = 'events'
+        for i, conditions in enumerate(physio_conditions):
+            print(conditions)
+            event_type = event_types_for_conditions[i]
+            if event_type == 'omissions':
+                response_window_duration = 0.75
+            else:
+                response_window_duration = 0.5
+            print('creating multi_session_df for', data_type, event_type, conditions)
+            try: # use try except so that it skips over any conditions that fail to generate for some reason
+                df = io.get_multi_session_df(project_code, session_number, conditions, data_type, event_type,
+                                             time_window=time_window, interpolate=interpolate, output_sampling_rate=output_sampling_rate,
+                                             response_window_duration=response_window_duration,
+                                             use_extended_stimulus_presentations=use_extended_stimulus_presentations,
+                                             overwrite=False)
+            except Exception as e:
```

Comment:
that's a nice trick, I was wondering how to add exceptions to code.

Reply:
learned it from Doug ;)